packages/cli/src/commands/dev.js

```diff
@@ -66,13 +55,32 @@ export default {
     // Start Build
     await builder.build()
 
+    if (!config.loadingScreen) {
+      this.listen(nuxt, argv)
+    }
+
     // Print memory usage
     showMemoryUsage()
 
     // Return instance
     return nuxt
   },
 
+  async listen(nuxt, argv) {
+    // Start listening
+    await nuxt.server.listen()
+
+    // Show banner when listening
+    showBanner(nuxt, false)
```

Comment:
Was the move of showBanner on purpose? I believe `Show banner when listening` is meant to be read as `showBanner after nuxt has started listening` and not `only showBanner when nuxt is listening`. Probably the showBanner call should be placed just after the call to this listen fn in _startDev

Reply:
Yes, `showBanner` should happen after `nuxt.server.listen`. These lines should move together. If I moved it into `_startDev` I would have to call `showBanner` in two places after calling `this.listen`.
I struggled to find a good name for `this.listen`. As you pointed out it doesn't just start listening, it also displays the banner and opens the page in the browser. These should all be in the same method not to duplicate code.

Reply:
Hmm, moving the listen call after the build is a second feature change and I actually missed that yesterday. Although I understand your use-case, I think it could be confusing that a seemingly unrelated toggle of loadingScreen also changes the point where Nuxt starts listening, @pi0 wdyt?

Also, when we would approve this we could just have showBanner print the memory usage on !loadingScreen instead of printing a separate line.

Reply:
Calling it `loadingScreen` at this stage can indeed be confusing. I wouldn't really want to call it `buildBeforeListen` or something since it wouldn't make sense to nuxt users. As a nuxt user I want an option to disable the loading screen which also happens to require starting to listen after building has completed.

Reply:
Please see my latest comment